### PR TITLE
compiler: Fix elastic after rebase

### PR DESCRIPTION
**Pull request by georgebisbas**
Branch to branch PR for passing elastic with emilien/stencil.store-result

### CODE REVIEW

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 1]
```diff
@@ -455,8 +461,10 @@ def generate_equations(self, eqs: list[Any], **kwargs):
         # Lower equations to their xDSL equivalent
         for eq in eqs:
             if isinstance(eq, Eq):
-                lowered = self.operator._lower_exprs(as_tuple(eq), **kwargs)[0]
-                self._convert_eq(lowered, **kwargs)
+                # Nested lowering? TO re-think approach
+                lowered = self.operator._lower_exprs(as_tuple(eq), **kwargs)
+                for lo in lowered:
+                    self._convert_eq(lo)
```
Comment (PapyChacal) [inline devito/ir/ietxdsl/cluster_to_ssa.py:467]: 👌

**Review by PapyChacal — APPROVED** [review]

**Comment by georgebisbas** [comment]
I am not merging it yet so I totally find my rebase bug locally.

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 2]
```diff
@@ -559,8 +568,16 @@ def convert(self, eqs: Iterable[Eq], **kwargs) -> builtin.ModuleOp:
             functions = OrderedSet()
             for eq in eqs:
                 if isinstance(eq, Eq):
-                    for f in retrieve_functions(eq):
+                    funcs = retrieve_function_carriers(eq)
+                    carriers = retrieve_function_carriers(eq)
+                    try:
+                        assert funcs == carriers
```
Comment (PapyChacal) [inline devito/ir/ietxdsl/cluster_to_ssa.py:574]: Is it supposed to use a different retriever? You're using the same function twice here

File: devito/ir/ietxdsl/cluster_to_ssa.py [hunk 3]
```diff
@@ -559,8 +568,16 @@ def convert(self, eqs: Iterable[Eq], **kwargs) -> builtin.ModuleOp:
...
+                    except AssertionError:
+                        raise AssertionError(f"Function carriers {carriers} do not match functions {funcs}")  # noqa
+
+                    for f in retrieve_function_carriers(eq):
```
Comment (PapyChacal) [inline devito/ir/ietxdsl/cluster_to_ssa.py:578]: Sounds like this one could reuse

**Review by PapyChacal — APPROVED** [review]
Thanks!!
Minor comments and GTG!

**Pull request merged**